dev/NavigationView/NavigationViewItemPresenter.cpp

```diff
@@ -54,4 +64,23 @@ NavigationViewItem* NavigationViewItemPresenter::GetNavigationViewItem()
         navigationViewItem = winrt::get_self<NavigationViewItem>(item);
     }
     return navigationViewItem;
-}
\ No newline at end of file
+}
+
+void NavigationViewItemPresenter::UpdateCompactPaneLength(double compactPaneLength, bool shouldUpdate)
+{
+    if (!this) {
+        // "We" are null, skip
+        return;
+    }
+
+    m_compactPaneLengthValue = compactPaneLength;
+    if (!shouldUpdate) {
+        return;
+    }
+    if (auto iconGridColumn = GetTemplateChildT<winrt::ColumnDefinition>(c_iconBoxColumnDefinitionName, *this))
```

Comment:
>GetTemplateChildT [](start = 30, length = 17)

This call is relatively expensive, Its unclear to me how often this will get called, does it make sense to store the result when we apply the template?

Reply:
I've tried that, however once the NavigationView switches the DisplayMode, we switch visual states and our reference get's lost (more specifically, accessing it results in an unauthorized access violation). And since apparently the items switch their visual states and template often, it is easier to get a fresh reference.

Reply:
Does it make more sense to get the ColumnDefinition from the grid instead of making it a named template part then?

Reply:
The problem is that the reason why we grab a fresh copy of the Column every time also applies to the Grid it is contained in. I don't think we get around asking for the TemplateChild in that function, and if why ask for it, why not ask for the Column directly, instead of asking for the Grid and then accessing the Grids ColumnDefinitions.